### ISSUE TITLE
The jsx-a11y/no-static-element-interactions rule doesn't work properly in this preset

### ISSUE THREAD

**Issue by callumlocke** [airbnb/javascript#1374]
Take this code (based on the [no-static-element-interactions example](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/no-static-element-interactions.md)):

```jsx
import React from 'react';

const fn = () => {};

const element = <div
  onClick={fn}
  onKeyPress={fn}
  role="button"
  tabIndex="0">
  Save
</div>
```

The above code should pass the `jsx-a11y/no-static-element-interactions` rule, because the element has `role="button"`, so it's OK.

And it _does_ pass that rule, when the rule is used on its own:

```yml
rules:
  jsx-a11y/no-static-element-interactions: error
```

But when the rule is used as part of the airbnb present, it fails, no matter what you put in the `role` attribute:

```yml
extends:
  - airbnb
```

```
Visible, non-interactive elements should not have mouse or
keyboard event listeners (jsx-a11y/no-static-element-interactions)
```

**Comment by ljharb** [comment]
This is a bug for eslint-plugin-jsx-a11y; not for this repo.

However, roles are supposed to be a last resort; that should really be a `<button>`.
